cli/metrics/conn_darwin.go

```diff
@@ -0,0 +1,43 @@
+//go:build darwin
+// +build darwin
+
+/*
+   Copyright 2022 Docker Compose CLI authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package metrics
+
+import (
+	"net"
+	"path/filepath"
+
+	"github.com/docker/docker/pkg/homedir"
+)
+
+var (
+	socket = "/var/run/docker-cli.sock"
+)
+
+func init() {
+	// Attempt to retrieve the Docker CLI socket for the current user.
+	if home := homedir.Get(); home != "" {
+		socket = filepath.Join(home, "/Library/Containers/com.docker.docker/Data/docker-cli.sock")
```

Comment:
Maybe we should use a shorter Linux-style path in `$HOME/.docker/desktop/docker-cli.sock` even on Darwin because we've had some problems where the maximum path in the sockaddr is 104 characters (even shorter than Linux's 108 characters). We work around this in some of our binaries by calling `os.Chdir("Library/Containers/com.docker.docker")` and then using a relative path... but I suspect this trick doesn't work for the CLI because changing the current directory probably breaks something.

WDYT?

Reply:
Hmm, this is a tough one.

The best thing I can think of from the top of my head is:

    pushd ...
    connect with relative path
    popd
    
but not sure that would be safe to do.

Reply:
Reflecting on it a bit, I'm not as concerned as I was because the socket is best-effort (and it will work probably 99% of the time). So I'm happy with it as-is.